modules/liquidity-pools/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+			Ok(())
+		}
+
+		pub fn set_enabled_trades(origin, pool_id: T::LiquidityPoolId, currency_id: T::CurrencyId, longs: Leverages, shorts: Leverages) -> Result {
+			let who = ensure_signed(origin)?;
+			Self::_set_enabled_trades(&who, pool_id, currency_id, longs, shorts)?;
+			Self::deposit_event(RawEvent::SetEnabledTrades(who, pool_id, currency_id, longs, shorts));
+			Ok(())
+		}
+	}
+}
+
+decl_error! {
+	// LiquidityPools module errors
+	pub enum Error {
+		NoPermission,
+		CannotCreateMorePool,
+		CannotRemovePool,
+		CannotDepositAmount,
+		CannotWithdrawAmount,
+	}
+}
+
+impl<T: Trait> Module<T> {
+	pub fn is_owner(pool_id: T::LiquidityPoolId, who: &T::AccountId) -> bool {
+		match Self::owners(pool_id) {
+			Some(id) => &id == who,
+			None => false,
+		}
+	}
+
+	pub fn is_enabled(pool_id: T::LiquidityPoolId, currency_id: T::CurrencyId, leverage: Leverage) -> bool {
+		match <LiquidityPoolOptions<T>>::get(&pool_id, &currency_id) {
+			Some(pool) => pool.enabled_longs.contains(leverage) || pool.enabled_shorts.contains(leverage),
+			None => false,
+		}
+	}
+}
+
+// Private methods
+impl<T: Trait> Module<T> {
+	fn _create_pool(who: &T::AccountId) -> result::Result<T::LiquidityPoolId, Error> {
+		let pool_id = Self::next_pool_id();
+		// increment next pool id
+		let next_pool_id = pool_id.checked_add(&One::one()).ok_or(Error::CannotCreateMorePool)?;
+		<NextPoolId<T>>::put(next_pool_id);
+		// owner reference
+		<Owners<T>>::insert(&pool_id, who);
+		Ok(pool_id)
+	}
+
+	fn _disable_pool(who: &T::AccountId, pool_id: T::LiquidityPoolId) -> result::Result<(), Error> {
+		ensure!(Self::is_owner(pool_id, who), Error::NoPermission);
+
+		for currency_id in T::LiquidityCurrencyIds::get() {
+			if let Some(mut pool) = <LiquidityPoolOptions<T>>::get(&pool_id, currency_id) {
+				pool.enabled_longs = Leverages::none();
+				pool.enabled_shorts = Leverages::none();
+				<LiquidityPoolOptions<T>>::insert(&pool_id, currency_id, pool);
+			}
+		}
+
+		Ok(())
+	}
+
+	fn _remove_pool(who: &T::AccountId, pool_id: T::LiquidityPoolId) -> result::Result<(), Error> {
+		ensure!(Self::is_owner(pool_id, who), Error::NoPermission);
+		ensure!(T::PoolManager::can_remove(pool_id), Error::CannotRemovePool);
+
+		let balance = Self::balances(&pool_id);
+		// deposit liquidity balance to pool owner
+		T::LiquidityCurrency::deposit(who, balance).map_err(|e| e.into())?;
+
+		<Balances<T>>::remove(&pool_id);
+		<Owners<T>>::remove(&pool_id);
+
+		for currency_id in T::LiquidityCurrencyIds::get() {
+			<LiquidityPoolOptions<T>>::remove(&pool_id, currency_id);
+		}
+
+		Ok(())
+	}
+
+	fn _deposit_liquidity(
+		who: &T::AccountId,
+		pool_id: T::LiquidityPoolId,
+		amount: T::Balance,
+	) -> result::Result<(), Error> {
+		let balance = Self::balances(&pool_id);
+		let new_balance = balance.checked_add(&amount).ok_or(Error::CannotDepositAmount)?;
+		// withdraw account
+		T::LiquidityCurrency::withdraw(who, amount).map_err(|e| e.into())?;
```

Comment:
Instead of use withdraw / deposit, which modifies total issuance, use use transfer from / into module account

Check this as example:
https://github.com/AcalaNetwork/Acala/blob/239eabe1957ddb2584d343b78aa75c485577ffe1/modules/auction_manager/src/lib.rs#L108

https://github.com/AcalaNetwork/Acala/blob/239eabe1957ddb2584d343b78aa75c485577ffe1/modules/auction_manager/src/lib.rs#L169